tests/ZendTest/Form/Element/DateSelectTest.php

```diff
@@ -83,4 +83,10 @@ public function testConstructAcceptsDayAttributes()
         $dayAttributes = $sut->getDayAttributes();
         $this->assertEquals('test', $dayAttributes['class']);
     }
+
+    public function testValueSetterReturnsSameObjectType()
+    {
+        $element  = new DateSelectElement();
+        $this->assertInstanceOf(get_class($element), $element->setValue('hello world'));
```

Comment:
<code>setValue('hello world')</code> must be end up with Exception because of this https://github.com/matwright/zf2/blob/master/library/Zend/Form/Element/DateSelect.php#L97-L101